src/per/qspi.cpp

```diff
@@ -330,39 +330,49 @@ QSPIHandle::Impl::Write(uint32_t address, uint32_t size, uint8_t* buffer)
 QSPIHandle::Result QSPIHandle::Impl::Erase(uint32_t start_addr,
                                            uint32_t end_addr)
 {
-    uint32_t block_addr;
-    uint32_t block_size = IS25LP080D_SECTOR_SIZE; // 4kB blocks for now.
-    // 64kB chunks for now.
-    start_addr = start_addr - (start_addr % block_size);
-    while(end_addr > start_addr)
+    if((end_addr - start_addr) >= IS25LP080D_BLOCK_SIZE)
     {
-        block_addr = start_addr & 0x0FFFFFFF;
-        if(EraseSector(block_addr) != QSPIHandle::Result::OK)
+        uint32_t block_addr;
+        uint32_t block_size = IS25LP080D_BLOCK_SIZE;
+        start_addr          = start_addr - (start_addr % block_size);
+        while(end_addr > start_addr)
         {
-            ERR_RECOVERY(Status::E_HAL_ERROR);
+            block_addr = start_addr & 0x0FFFFFFF;
+            if(EraseCmd(block_addr, BLOCK_ERASE_CMD) != QSPIHandle::Result::OK)
+            {
+                ERR_RECOVERY(Status::E_HAL_ERROR);
+            }
+            start_addr += block_size;
+        }
```

Comment:
> I have added the following constrain to the if block ((end_addr - start_addr) % IS25LP080D_BLOCK_SIZE) == 0.
Thus the BLOCK_ERASE should only take place if the memory area is exactly multiple of BLOCK_SIZE

This definitely mitigates the over-erase issue, but it also means that unusually sized erases will still be slow. As a compromise, what about moving the size check to determine block vs sector erase to _inside_ the `while` loop, so we're comparing the remaining erase range each time rather than the entire original range, to decide whether to erase a block or sector?

> Personally, I would prefer to tag the Erase method OBSOLETE, and replace it with a public EraseCmd.

I can see where you're coming from. However given that would be a breaking change I would imagine that would require a little more review and caution.

> By the way - the fix_style.sh makes too many modifications in my environment - which in turn are reported in the github CI.

If I remember correctly this is a known problem (happens to me too). Don't worry about it, as long as the files your changes are in pass the lint check we're good.

Reply:
As a companion to the more generalized public `Erase` method and the existing public `EraseSector` method, there's also the option of adding a public `EraseBlock` method that calls into `EraseCmd` to erase the whole block at the specified offset.

Reply:
Ok, an public `EraseBlock ` method would also be a good alternative. The `Erase `method, in general, is and was a bit dangerous - the address range had to correspond to the 4KB sectors, and could not be arbitrary...

Reply:
@ndonald2,

thanks for your feedback.

I just looked at the EraseSector method again - actually any address checking is missing there. If we want to introduce an additional EraseBlock method, we have to consider that there is a BLOCK_ERASE_CMD (64KB) and a BLOCK_ERASE_32K_CMD. In short, I want to contribute only improvements that I really use and where I can say that it works - all other special cases with arbitrary addresses I do not want to handle.